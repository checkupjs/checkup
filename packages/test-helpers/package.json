{
  "name": "@checkup/test-helpers",
  "description": "Checkup's test helper library",
  "version": "0.0.10",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {
    "@checkup/core": "0.0.10",
    "fixturify-project": "^2.1.0",
    "handlebars": "^4.7.6",
    "stdout-stderr": "^0.1.13",
    "tmp": "^0.2.0"
  },
  "engines": {
    "node": ">= 12.*"
  },
  "files": [
    "/lib",
    "/static"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc --build",
    "test": "jest --passWithNoTests --no-cache"
  },
  "types": "lib/index.d.ts"
}
