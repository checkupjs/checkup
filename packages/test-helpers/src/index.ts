export { default as Project } from './ember-cli-fixturify-project';
export { stdout } from './stdout';
export { default as CheckupProject } from './checkup-fixturify-project';
export { default as Plugin } from './plugin';
