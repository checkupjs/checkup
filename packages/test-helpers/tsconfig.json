{
  "extends": "../../tsconfig-base.json",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src",
    "types": ["chai", "jest", "node"],
    "esModuleInterop": true
  },
  "references": [
    {
      "path": "../core"
    }
  ],
  "include": ["src/**/*"]
}
