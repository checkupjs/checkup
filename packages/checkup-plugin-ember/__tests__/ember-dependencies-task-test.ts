import { EmberProject, stdout, getTaskContext } from '@checkup/test-helpers';

import EmberDependenciesTask from '../src/tasks/ember-dependencies-task';
import EmberDependenciesTaskResult from '../src/results/ember-dependencies-task-result';

describe('dependencies-task', () => {
  let emberProject: EmberProject;

  beforeEach(function () {
    emberProject = new EmberProject('checkup-app', '0.0.0', (project) => {
      project.addDependency('ember-source', '^3.15.0');
      project.addDependency('ember-cli', '^3.15.0');
      project.addDevDependency('ember-cli-string-utils', 'latest');
    });

    emberProject.addAddon('ember-cli-blueprint-test-helpers', 'latest');
    emberProject.writeSync();
  });

  afterEach(function () {
    emberProject.dispose();
  });

  it('detects Ember dependencies', async () => {
    const result = await new EmberDependenciesTask(
      getTaskContext({ path: emberProject.baseDir })
    ).run();
    const dependencyTaskResult = <EmberDependenciesTaskResult>result;

    dependencyTaskResult.toConsole();

    expect(stdout()).toMatchSnapshot();
  });

  it('detects Ember dependencies as JSON', async () => {
    const result = await new EmberDependenciesTask(
      getTaskContext({ path: emberProject.baseDir })
    ).run();
    const dependencyTaskResult = <EmberDependenciesTaskResult>result;

    expect(dependencyTaskResult.toJson()).toMatchSnapshot();
  });

  it('detects Ember dependencies for html, and doesnt create a table without dependencies', async () => {
    const result = await new EmberDependenciesTask(
      getTaskContext({ path: emberProject.baseDir })
    ).run();
    const dependencyTaskResult = <EmberDependenciesTaskResult>result;
    const htmlResults = dependencyTaskResult.toReportData();

    expect(htmlResults).toMatchSnapshot();
    expect(htmlResults).toHaveLength(4);
  });
});
