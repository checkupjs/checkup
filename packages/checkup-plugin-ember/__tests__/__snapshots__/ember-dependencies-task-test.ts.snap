// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dependencies-task detects Ember dependencies 1`] = `
"=== Ember Dependencies ===

Dependency Types                 Total 
Ember Core Libraries             3     
Ember Addon Dependencies         1     
Ember Addon Dev Dependencies     0     
Ember CLI Addon Dependencies     2     
Ember CLI Addon Dev Dependencies 1     

"
`;

exports[`dependencies-task detects Ember dependencies as JSON 1`] = `
Object {
  "meta": Object {
    "friendlyTaskName": "Ember Dependencies",
    "taskClassification": Object {
      "category": "insights",
      "priority": "high",
    },
    "taskName": "ember-dependencies",
  },
  "result": Array [
    Object {
      "data": Object {
        "ember-cli": "^3.15.0",
        "ember-data": "Not found",
        "ember-source": "^3.15.0",
      },
      "displayName": "Ember Core Libraries",
      "total": 3,
      "type": "ember core libraries",
    },
    Object {
      "data": Object {
        "ember-source": "^3.15.0",
      },
      "displayName": "Ember Addon Dependencies",
      "total": 1,
      "type": "ember addon dependencies",
    },
    Object {
      "data": Object {},
      "displayName": "Ember Addon Dev Dependencies",
      "total": 0,
      "type": "ember addon devDependencies",
    },
    Object {
      "data": Object {
        "ember-cli": "^3.15.0",
        "ember-cli-blueprint-test-helpers": "latest",
      },
      "displayName": "Ember CLI Addon Dependencies",
      "total": 2,
      "type": "ember-cli addon dependencies",
    },
    Object {
      "data": Object {
        "ember-cli-string-utils": "latest",
      },
      "displayName": "Ember CLI Addon Dev Dependencies",
      "total": 1,
      "type": "ember-cli addon devDependencies",
    },
  ],
}
`;

exports[`dependencies-task detects Ember dependencies for html, and doesnt create a table without dependencies 1`] = `
Array [
  TableData {
    "componentType": "table",
    "formattedResult": Array [
      Object {
        "result": Object {
          "name": "ember core libraries",
          "value": 3,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "ember addon dependencies",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "ember addon devDependencies",
          "value": 0,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "ember-cli addon dependencies",
          "value": 2,
        },
        "rowClass": "bg-gray-100",
      },
      Object {
        "result": Object {
          "name": "ember-cli addon devDependencies",
          "value": 1,
        },
        "rowClass": "bg-gray-100",
      },
    ],
    "meta": Object {
      "friendlyTaskName": "Ember Dependencies",
      "taskClassification": Object {
        "category": "insights",
        "priority": "high",
      },
      "taskName": "ember-dependencies",
    },
    "tableHeaders": Array [
      "name",
      "value",
    ],
  },
]
`;
